src/uu/wc/src/wc.rs

```diff
@@ -306,29 +306,83 @@ fn word_count_from_reader<T: WordCountable>(
     mut reader: T,
     settings: &Settings,
 ) -> (WordCount, Option<io::Error>) {
-    let only_count_bytes = settings.show_bytes
-        && (!(settings.show_chars
-            || settings.show_lines
-            || settings.show_max_line_length
-            || settings.show_words));
-    if only_count_bytes {
-        let (bytes, error) = count_bytes_fast(&mut reader);
-        return (
-            WordCount {
-                bytes,
-                ..WordCount::default()
-            },
-            error,
-        );
-    }
-
-    // we do not need to decode the byte stream if we're only counting bytes/newlines
-    let decode_chars = settings.show_chars || settings.show_words || settings.show_max_line_length;
-
-    if !decode_chars {
-        return count_bytes_and_lines_fast(&mut reader);
+    match (
+        settings.show_bytes,
+        settings.show_chars,
+        settings.show_lines,
+        settings.show_max_line_length,
+        settings.show_words,
+    ) {
+        (false, false, false, false, false) => (WordCount::default(), None),
+        (true, false, false, false, false) => {
+            // Fast path when only show_bytes is true.
+            let (bytes, error) = count_bytes_fast(&mut reader);
+            (
+                WordCount {
+                    bytes,
+                    ..WordCount::default()
+                },
+                error,
+            )
+        }
+        (false, false, true, false, false) | (true, false, true, false, false) => {
+            // Fast path when only (show_bytes || show_lines) is true.
+            count_bytes_and_lines_fast(&mut reader)
+        }
+        (_, false, false, false, true) => {
+            word_count_from_reader_specialized::<_, false, false, false, true>(reader)
+        }
+        (_, false, false, true, false) => {
+            word_count_from_reader_specialized::<_, false, false, true, false>(reader)
+        }
+        (_, false, false, true, true) => {
+            word_count_from_reader_specialized::<_, false, false, true, true>(reader)
```

Comment:
The code coverage indicates that we don't have a test covering this (and at a few more occurrences below).
Would it be possible to add tests covering this?
thanks

Reply:
Done